js/bundle.go

```diff
@@ -60,7 +60,16 @@ type BundleInstance struct {
 
 // NewBundle creates a new bundle from a source file and a filesystem.
 func NewBundle(src *loader.SourceData, filesystems map[string]afero.Fs, rtOpts lib.RuntimeOptions) (*Bundle, error) {
-	compiler, err := compiler.New()
+	compatMode := compiler.CompatibilityModeES6
+	if rtOpts.CompatibilityMode.Valid {
+		if cm, err := lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String); err == nil {
```

Comment:
I think checking for error first looks better:

```
cm, err := lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String)
if err != nil {
    return nil, err
}
compatMode = cm
```

Reply:
Since an empty string returns as invalid (maybe it shouldn't?), this needs to be a slightly longer:

```go
	compatMode := compiler.CompatibilityModeES6
	var cm compiler.CompatibilityMode
	if rtOpts.CompatibilityMode.Valid {
		var err error
		cm, err = lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String)
		if err != nil {
			return nil, err
		}
	}
	compatMode = cm
```

I'm not sure if that's more readable over an `else` block. Let me know.

Reply:
Resolved by 20aa6f5a.